lib/intercom/scroll_collection_proxy.rb

```diff
@@ -0,0 +1,83 @@
+require "intercom/utils"
+require "ext/sliceable_hash"
+
+module Intercom
+  class ScrollCollectionProxy
+
+    attr_reader :resource_name, :scroll_url, :resource_class, :scroll_param, :users
+
+    def initialize(resource_name, finder_details: {}, client:)
+      @resource_name = resource_name
+      @resource_class = Utils.constantize_resource_name(resource_name)
+      @scroll_url = (finder_details[:url] || "/#{@resource_name}") + '/scroll'
+      @client = client
+
+    end
+
+    def next(scroll_paramater=nil)
+      @users = []
```

Comment:
The companies endpoint also throws the `page limit reached, please use scroll API` error - thoughts on renaming this @records so it can handle both models?